app/src/main/java/com/vitorpamplona/amethyst/ui/components/ZoomableImageView.kt

```diff
@@ -57,29 +57,32 @@ fun ZoomableImageView(word: String) {
   )
 
   if (dialogOpen) {
-    Dialog(
-      onDismissRequest = { dialogOpen = false },
-      properties = DialogProperties(usePlatformDefaultWidth = false)
-    ) {
-      Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colors.background) {
-        Column(
-          modifier = Modifier.padding(10.dp)
-        ) {
-          Row(
-            modifier = Modifier
-              .fillMaxWidth(),
-            horizontalArrangement = Arrangement.SpaceBetween,
-            verticalAlignment = Alignment.CenterVertically
-          ) {
-            CloseButton(onCancel = {
-              dialogOpen = false
-            })
+    ZoomableImageDialog(word, onDismiss = { dialogOpen = false })
+  }
+}
 
-            SaveToGallery(url = word)
-          }
+@Composable
+fun ZoomableImageDialog(imageUrl: String, onDismiss: () -> Unit) {
+  Dialog(
+    onDismissRequest = onDismiss,
+    properties = DialogProperties(usePlatformDefaultWidth = false)
+  ) {
+    Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colors.background) {
+      Column(
+        modifier = Modifier.padding(10.dp)
+      ) {
+        Row(
+          modifier = Modifier
+            .fillMaxWidth(),
+          horizontalArrangement = Arrangement.SpaceBetween,
+          verticalAlignment = Alignment.CenterVertically
+        ) {
+          CloseButton(onCancel = onDismiss)
 
-          ZoomableAsyncImage(word)
+          SaveToGallery(url = imageUrl)
         }
+
+        ZoomableAsyncImage(imageUrl)
```

Comment:
This logic was previously in `ZoomableImageView`, extracted into a separate `ZoomableImageDialog` for reuse.